File: frontend/src/types/plan.ts
import type { PageQueryParams, Status } from "./common";

export interface Plan {
  id: number;
  name: string;
  price: string;
  duration_days: number;
  status: Status;
  created_at: string;
  updated_at: string;
}

export interface PlanPayload {
  name: string;
  price: string;
  duration_days: number;
  status: Status;
}

export interface PlanQueryParams extends PageQueryParams {
  name?: string;
}

File: backend/app/models/plan.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import DateTime, Enum, Numeric, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.enums import PlanStatus
from app.database.base import Base


class Plan(Base):
    __tablename__ = "plans"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), unique=True, index=True, nullable=False)
    price: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    duration_days: Mapped[int] = mapped_column(nullable=False)
    status: Mapped[PlanStatus] = mapped_column(
        Enum(PlanStatus, native_enum=False, length=20),
        nullable=False,
        default=PlanStatus.ACTIVE,
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )
